mealpy.py

```diff
@@ -27,31 +29,35 @@
 
 KEYRING_SERVICENAME = BASE_DOMAIN
 
-# TODO(GH-4): Replace this feature toggle with cli option
-USE_KEYRING = False
-
 
 def load_config():
     schema = strictyaml.Map({
         'email_address': strictyaml.Email(),
+        'use_keyring': strictyaml.Bool()
     })
-
     root_dir = path.abspath(path.dirname(__file__))
-    with open(path.join(root_dir, 'config.yaml')) as config_file:
+    fname = path.join(root_dir, 'config.yaml')
+    # Create new file using template if not already existing
+    if not path.isfile(fname):
+        template = path.join(root_dir, 'config.template.yaml')
+        copyfile(template, fname)
+    with open(fname) as config_file:
         return strictyaml.load(config_file.read(), schema).data
 
+CONFIG = load_config()
```

Comment:
Can we avoid import time side-effects? It makes testing difficult because `from mealpal import MealPal` does more than expected, when we only need to make the class available for unit testing.

Reply:
Should we pull out the `MealPal` class into its own file then? The CLI part and the `MealPal` class don't really need to be in the same file.

Reply:
My issue is with import-time side-effects. which is bad-practice (imo). Moving `MealPal` or `CONFIG` to a different module is not fixing the issue, just limiting its impact.

Perhaps a more direct and illustrative example will help frame this discussion: we might have a helper method for quasi-deserializing multiple config values into an object. This helper method is pure and stateless, making it super-testable. However, in order to unit test this helper, I would still need to mock out the file IO calls to prevent unit tests from modifying real files on disk. Simply because the act of `import  config` causes lots of code to execute.

Why rename `load_config` to `get_config` and call the function instead of reference a global const? Throw an lru_cache on it and it's identical behaviour.

Reply:
I'm open to changing this, but in the example you mentioned, if `load_config` were to be moved to a helper file, wouldn't the test file just do `from some_helper_file import load_config`, and the function would be imported without any side effects? 

Unless you're referring to the side effect of file creation within `load_config`, which can be extracted outside and ran before `load_config` itself.

Reply:
Whatever module that has `CONFIG = load_config()` will be a problem to test. The issue is not about where `load_config` lives.

It's not enough to isolate these import-time side-effects to a separate module because that doesn't solve the problem, only hides it in the corner. The goal is to not only allow me to unit test `load_config`, but to also have 100% test coverage. I will want to test every module and the code in the corner will eventually give me trouble.

imo, I should be able to import any python module (in the world) and not need to worry about code being needlessly being executed. An example of needless if I called `from config import KEYRING_SERVICE_NAME`, I expect a scalar, static value is imported into the namespace and there's no reason to read a config file on disk or make a copy in working dir, etc.

Reply:
Reverting back to simply `load_config`.